web/js/components/context-menu/context-menu.js

```diff
@@ -68,13 +74,43 @@ function RightClickMenu(props) {
     };
   };
 
+  let longPressTimer = null;
+
+  function setTimer(incomingEvent, incomingOlMap) {
+    longPressTimer = setTimeout(() => {
+      setIsHoldPress(true);
+      handleContextEvent(incomingEvent, incomingOlMap);
+    }, 1000);
+    return longPressTimer;
+  }
+
+  function handleTouchStart(event, olMap) {
+    event.pixel = [event.touches[0].clientX, event.touches[0].clientY];
+    setTimer(event, olMap);
+  }
+
+  function clearTimer() {
+    clearTimeout(longPressTimer);
+  }
+
+  function handleDrag() {
+    setShow(false);
+  }
+
   useEffect(() => {
     if (isCoordinateSearchActive) return;
     events.on('map:singleclick', handleClick);
     events.on('map:contextmenu', handleContextEvent);
+
+    events.on('map:touchstart', handleTouchStart);
+    events.on('map:touchend', clearTimer);
+    events.on('map:drag', handleDrag);
```

Comment:
Maybe not a huge deal but, I noticed that this considers any mouse movement, with either left or right click down, to be a drag event.  That means that, if you move the mouse even slightly while right-clicking to show the context menu that it immediately gets hidden again.  On desktop this probably isn't likely but I'm wondering if this is a concern on mobile where it may be more likely that a user slightly moves their finger while doing a long press.  I wonder if we can listen for a different event that would only fire when a left-click + drag event occurs.